Exercise_4.java

```diff
@@ -31,10 +32,34 @@ static void inorder(Node temp)
        
     /*function to insert element in binary tree */
     static void insert(Node temp, int key) 
-    { 
-
-        // Do level order traversal until we find 
-        // an empty place and add the node.  
+    {   
+        Node n = new Node(key);
+        Queue q = new LinkedList(); //Using the queue for keepig the track of left and right nodes, so that we can traverse in a level wise
+        Node dummy = null;   //Using this dummy to mark that current level has reached end.

```

Comment:
Can be done without using this dummy node
Look for level order traversal of binary tree online